### PR TITLE
replication: limit larger uploads to a subset of workers

### PR DESCRIPTION
**Pull request by poornas**
## Motivation and Context
Limit large uploads (> 128MiB)  to a max of 10 workers, intent is to avoid larger uploads from using all replication bandwidth, giving room for smaller uploads to sync faster.

## How to test this PR?
Nothing that affects functionality

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Review by harshavardhana — APPROVED** [review]

File: cmd/bucket-replication.go [hunk 1]
```diff
@@ -1861,16 +1865,34 @@ func (p *ReplicationPool) ResizeFailedWorkers(n int) {
...
 	h := xxh3.HashString(bucket + object)
 	p.mu.RLock()
 	defer p.mu.RUnlock()
-	if len(p.workers) == 0 {
+	workerCnt := uint64(len(p.workers))
+	if workerCnt == 0 {
 		return nil
 	}
-	return p.workers[h%uint64(len(p.workers))]
+	largeWorkerCount := uint64(math.Min(maxLargeWorkers, float64(sz/minLargeObjSize)))
```
Comment (klauspost) [inline cmd/bucket-replication.go]: This is soo complicated. My head hurts ;)

Can't we just allocate 10 workers separately and use these for big objects exclusively?

Comment (poornas) [reply cmd/bucket-replication.go]: trying to avoid creating yet more workers and managing their auto scaling separately

Comment (klauspost) [reply cmd/bucket-replication.go]: @poornas Can't we just create 10 static workers?

Comment (poornas) [reply cmd/bucket-replication.go]: we could. changed the PR

**Review by kannappanr — APPROVED** [review]
LGTM

**Pull request merged**